### ISSUE TITLE
map_identity: false negative with map_err

### ISSUE THREAD

**Issue by TonalidadeHidrica** [rust-lang/rust-clippy#8478]
### What it does

It detects verbose identity conversions like `iterator.map(|a| a)`, `result.map(|a| a)`, `result.map_err(|a| a)`, `option.map(|a| a)` and so on.

### Lint Name

needless_identity_map

### Advantage

- Removes unnecessary code.

### Drawbacks

- Possibly eliminates readability originally intended by verbose code...?  I doubt it.

### Example

```rust
  let r: Result<u32, u32> = Ok(1);
  let u = r.map_err(|a| a);
```

Could be written as:

```rust
  let r: Result<u32, u32> = Ok(1);
  let u = r;
```

**Comment by matthiaskrgr** [comment]
Isn't this what the `map_identity` lint is about? :thinking: 

````rust
pub fn main() {
    let _ = vec![1, 2, 3].into_iter().map(|x| x).next();
}
````

````
warning: unnecessary map of the identity function
 [--> src/main.rs:2:38](https://play.rust-lang.org/#)  |
2 |     let _ = vec![1, 2, 3].into_iter().map(|x| x).next();
  |                                      ^^^^^^^^^^^ help: remove the call to `map`
  |
  = note: `#[warn(clippy::map_identity)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_identity
````

**Comment by TonalidadeHidrica** [comment]
@matthiaskrgr Sorry that I didn't check thoroughly.  I realized that it won't work for `Result::map_err` and submitted an over-generalized request.  I now confirmed that `Result::map_err` is not covered by `map_identity` lint.  Should I open a new issue?

**Comment by matthiaskrgr** [comment]
Nah, I can simply reopen this one. :smile: 
I think linting on `map_err` makes sense for the lint so that's a false negative :upside_down_face:
